### DISCUSSION TITLE
Retrieve dual variables of a qp problem solved via osqp

### DISCUSSION THREAD

**Discussion by MarcFacerias** [casadi/casadi#3426] [Q&A]
Hello, 

When trying to retrieve dual variables using opti.dual() syntaxis I only get Nans, I assume that's due to the feature not being implemented for conic solvers , do you know how could I retrieve those values from the osqp solution object? 

Thanks a lot, 
Marc

**Comment by sandeep026** [comment]
I am using 3.6.3  and had the same issue.  @jgillis , is the issue resolved?

https://github.com/casadi/casadi/discussions/3106

**Comment by MarcFacerias** [comment]
I'm closing this discussion as right now (3.6.3) the opti interface does not include this feature, you can access to that data coding the problem using either high or low level interfaces
